cas-client-core/src/main/java/org/jasig/cas/client/session/SingleSignOutHandler.java

```diff
@@ -298,18 +295,24 @@ private void destroySession(final HttpServletRequest request) {
 
         final String token = XmlUtils.getTextForElement(logoutMessage, "SessionIndex");
         if (CommonUtils.isNotBlank(token)) {
-            final HttpSession session = this.sessionMappingStorage.removeSessionByMappingId(token);
+            final ClientSession session = this.sessionMappingStorage.removeSessionByMappingId(token);
 
             if (session != null) {
                 final String sessionID = session.getId();
                 logger.debug("Invalidating session [{}] for token [{}]", sessionID, token);
 
                 try {
                     session.invalidate();
-                } catch (final IllegalStateException e) {
+                } catch (final RuntimeException e) {
                     logger.debug("Error invalidating session.", e);
                 }
-                this.logoutStrategy.logout(request);
+
+                try {
+                    request.logout();
```

Comment:
My analysis was incorrect. I missed that you moved LogoutStrategy and related logic into DelegatingHttpRequest.